drivers/i2c/i2c_dw.c

```diff
@@ -714,7 +720,12 @@ static int i2c_dw_transfer(const struct device *dev,
 		}
 
 		/* Wait for transfer to be done */
-		k_sem_take(&dw->device_sync_sem, K_FOREVER);
+		ret = k_sem_take(&dw->device_sync_sem, K_MSEC(CONFIG_I2C_DW_RW_TIMEOUT_MS));
+		if (ret != 0) {
+			write_intr_mask(DW_DISABLE_ALL_I2C_INT, reg_base);
+			value = read_clr_intr(reg_base);
+			break;
```

Comment:
Suggestion: This seems duplicate of the code below. May we refactor both so we enforce DRY?
https://github.com/zephyrproject-rtos/zephyr/blob/cac70373aa81b9989905ac530b9773e4b748e409/drivers/i2c/i2c_dw.c#L346-L353.

Reply:
Thanks for approval and suggestion. Does creating a static inline function to disable and clear all the I2C interrupts a good way to achieve DRY? Thanks!

Reply:
Sounds good to me!